gateway-protocol/src/main/proto/gateway.proto

```diff
@@ -403,6 +403,41 @@ message SetVariablesResponse {
   int64 key = 1;
 }
 
+message ModifyProcessInstanceRequest {
+  // the key of the process instance that should be modified
+  int64 processInstanceKey = 1;
+  // instructions describing which elements should be activated in which scopes,
+  // and which variables should be created
+  repeated ModificationActivateInstruction activateInstructions = 2;
+  // instructions describing which elements should be terminated
+  repeated ModificationTerminateInstruction terminateInstructions = 3;
+}
+
+message ModificationActivateInstruction {
+  // the id of the element that should be activated
+  string elementId = 1;
+  // the key of the ancestor scope the element instance should be created in
+  int64 ancestorElementInstanceKey = 2;
+  // instructions describing which variables should be created
+  repeated ModificationVariableInstruction variableInstructions = 3;
+}
+
+message ModificationVariableInstruction {
+  // the variables that should be created
+  string variables = 1;
+  // the id of the element in which scope the variables should be created
+  string scopeId = 2;
```

Comment:
🔧 This field is also optional, so users can set it to `-1`. Let's describe what happens in that case. 

>(optional) the element id of the scope where the variables should be created in, otherwise the variables are global for the process instance

See https://github.com/camunda/zeebe/issues/9638